packages/frontend-shared/src/store/login-connect-store.ts

```diff
@@ -0,0 +1,134 @@
+import type { BannersState } from '@packages/types'
+import { defineStore } from 'pinia'
+
+interface LoginUserData {
+  fullName: string | null
+  email: string | null
+}
+
+export interface LoginConnectState {
+  isLoginConnectOpen: boolean
+  utmMedium: string
+  isLoggedIn: boolean
+  isProjectConnected: boolean
+  isConfigLoaded: boolean
+  isOrganizationLoaded: boolean
+  isMemberOfOrganization: boolean
+  hasNoRecordedRuns: boolean
+  error: boolean
+  hasRecordedRuns: boolean
+  loginError: boolean
+  userData?: LoginUserData
+  firstOpened?: number
+  latestBannerShownTime?: number
+  hasNonExampleSpec: boolean
+  promptsShown: {
+    ci1?: number
+    loginModalRecord?: number
+  }
+  bannersState: BannersState
+}
+
+// The user can be in only one status at a time.
+// These are specifically related to the dashboard
+// and the progress from logging in to recording a run.
+export const userStatuses = [
+  'isLoggedOut',
+  'needsOrgConnect',
+  'needsProjectConnect',
+  'needsRecordedRun',
+  'allTasksCompleted',
+] as const
+
+export type UserStatus = typeof userStatuses[number]
+
+export type LoginConnectBooleanField = 'isLoggedIn' | 'isProjectConnected' | 'isConfigLoaded' | 'isOrganizationLoaded' | 'isMemberOfOrganization' | 'hasRecordedRuns' | 'hasNoRecordedRuns' | 'hasNonExampleSpec'
+
+export const useLoginConnectStore = defineStore({
+  id: 'loginConnect',
+
+  state (): LoginConnectState {
+    return {
+      isLoginConnectOpen: false,
+      utmMedium: '',
+      isLoggedIn: false,
```

Comment:
Should we pull this login modal state out into its own store and rename this store to just `connect-store`?   Looking at it more, it seems like we are combining two different needs into this one store.

Reply:
I think that makes sense but going to give it a little more thought. We can remove it from this PR though, since it's not directly used here, and look at the ergonomics of a different store in the branch that implements the login flow. It's probably fine to separate them and have a small "LoginConnectModal" store.

Reply:
I've ended up deciding that we should leave this login modal stuff in here. Given how closely some of these things are related - the UTM params, the tooltips that close when the login modal opens, maybe some other things, where I think it will be easier to keep track in one place. 

Based on Zach's comment I structured the states a bit better, into `user` and `project`, with just a few top-level states remaining. This feels less cluttered and I think it's fine for now.